cfgov/v1/tests/models/test_filterable_list_mixins.py

```diff
@@ -46,6 +46,16 @@ def test_get_context_raises_exception_for_super_obj_has_no_get_context(
             request=self.factory.get("/")
         )
 
+    @override_settings(FLAGS={"HIDE_ARCHIVE_FILTER_OPTIONS": [("boolean", True)]})
+    def test_get_context_raises_exception_for_super_obj_has_no_archive_context(
+        self
+    ):
+        self.assertRaises(
+            AttributeError,
+            self.mixin.get_context,
+            request=self.factory.get("/")
+        )
```

Comment:
I don't think this test is going to touch the `flag_enabled` conditional. 

At a glance, it actually doesn't look like we have a test that directly tests the `get_context` method, but you could add two that check that `has_archived_posts` is `False` even when the queryset has objects that meet the condition, and then another test that sets has_archived_posts as expected.

Reply:
Paired with @Scotchester to directly test the `get_context` method.